crypto/elgamal/ciphertext.go

```diff
@@ -58,51 +59,46 @@ func (z *Ciphertext) Add(x, y *Ciphertext) *Ciphertext {
 }
 
 // Serialize returns a slice of len 4*32 bytes,
-// representing the C1.X, C1.Y, C2.X, C2.Y as little-endian.
+// representing the C1.X, C1.Y, C2.X, C2.Y as little-endian,
+// in reduced twisted edwards form.
 func (z *Ciphertext) Serialize() []byte {
-	x1, y1 := z.C1.Point()
-	x2, y2 := z.C2.Point()
 	var buf bytes.Buffer
-	for _, bi := range []*big.Int{
-		x1,
-		y1,
-		x2,
-		y2,
-	} {
+	c1x, c1y := format.FromTEtoRTE(z.C1.Point())
```

Comment:
TODO: handle this at Curve level

Reply:
i'm not sure about this one, since Point() is according to docs exactly supposed to return Twisted Edwards coords, not RTE. but we'll sort this out